File: frontend-angular/src/app/models/produto.ts
import { Ingrediente } from "./ingrediente"


export type ProdutoBodyRequest = {
  loja_uuid: string,
  categoria_uuid: string,
  nome: string,
  preco: number,
  descricao: string,
  image_bytes: string
}

export type ProdutoPUTRequest = {
    nome: string,
    descricao: string,
    preco: number,
}

export type ProdutoResponse = {
  uuid: string
  loja_uuid: string,
  categoria_uuid: string,
  nome: string,
  preco: number,
  descricao: string,
  categoria?: any,
  ingredientes?: Array<Ingrediente>,
  image_url: string,
  preco_hoje?: number
}

File: backend/src/models/produto.py
from pydantic import BaseModel
from typing import Optional, List
from src.models import Preco


class Produto(BaseModel):
    __tablename__ = "produtos"
    nome: str
    descricao: str
    preco: float
    categoria_uuid: str
    loja_uuid: str

    uuid: Optional[str] = None


class ProdutoPOST(BaseModel):
    __tablename__ = "produtos"
    nome: str
    descricao: str
    preco: float
    categoria_uuid: str
    loja_uuid: str
    image_bytes: str
    filename: str

    uuid: Optional[str] = None


class ProdutoPUT(BaseModel):
    __tablename__ = "produtos"
    nome: str
    descricao: str
    preco: float


class ProdutoGET(BaseModel):
    __tablename__ = "produtos"
    nome: str
    descricao: str
    preco: float
    categoria_uuid: str
    loja_uuid: str
    precos: List[Preco]

    image_url: Optional[str] = None
    preco_hoje: Optional[float] = None
    uuid: Optional[str] = None
